chore: MSW 초기 환경 설정(#14)

KoCeleste opened the pull request:
<!-- PR 제목 규칙:
[type]: 작업 요약 (#이슈번호)
예시: feat: 로그인 페이지 UI 구현 (#12)
-->

## #️⃣ 연관된 이슈
- closes #14

## 📑 구현 사항
- MSW 패키지 설치
- 핸들러( GET / POST / PUT / DELETE ) 기본 구현
- browser.ts 파일 생성 및 worker 설정
- main.tsx에 개발 환경에서만 worker.start() 적용
- mockServiceWorker.js 파일을 public/ 경로에 배치
- 테스트 코드는 PR 전 삭제 완료
- .env.example 파일 추가

## 🌀 어려웠던 점 / 고민했던 부분
- MSW를 처음 써봐서 조금 오래 걸렸습니다.. ㅎㅎ..

## 📸 구현 이미지 (선택)
- 
<img width="1920" height="944" alt="mock api 테스트" src="https://github.com/user-attachments/assets/ff3dae16-06b0-4bc1-8435-34a150f927c3" />

## ❇️ 코드 설명 (선택)
- 

## 💬 리뷰 요청사항 (선택)
> 리뷰어에게 피드백을 받고 싶은 지점이 있다면 작성해주세요.

KoCeleste committed:
chore: MSW 초기 환경 설정

- handlers.ts TypeScript 버전 생성
- browser.ts 파일 생성 및 setupWorker 설정
- main.tsx enableMocking 적용
- mockServiceWorker.js public 폴더에 배치
- TypeScript 오류 해결 및 경로 정리

agrade1 reviewed (dismissed):
MSW 초기설정 잘 작성 해주신거 같습니다!
.env 파일 환경변수 추가 관련 내용 추가도 해주세요

yeeun-kor commented on src/main.tsx:
```diff
@@ -1,12 +1,24 @@
-  </StrictMode>
-)
+async function enableMocking() {
+  if (process.env.NODE_ENV !== 'development') {
+    return
+  }
+
```
vite에서 env환경설정 변수명 불러올때, 
'import.mete.VITE_어쩌고저쩌고' 로 알고있습니다만,

yeeun-kor replied:
이 부분 확인이 필요할 것 같습니다./

agrade1 replied:
NEXT_PUBLIC_API_MOCKING 이건 제가 넥스트 작업했을때 msw 사용하는 환경변수 네이밍이였습니다

KoCeleste replied:
말씀하신 코드 수정하고 .env.example 파일 추가했습니다 확인부탁드려욤

KoCeleste committed:
chore: 환경변수 템플릿 파일 추가

- .env.example 파일 생성하여 팀원들이 환경 변수 구성을 참고할 수 있도록 제공
- .env 파일은 .gitignore에 포함하여 버전 관리 제외함

KoCeleste committed:
chore: .env.example 템플릿 내용 추가

- 환경 변수 샘플 값 추가하여 팀원이 참고 가능하도록 수정

agrade1 reviewed (approved):
변수 네이밍 잘 적용된거 같습니다.
